### ISSUE TITLE
Design rationale for TiledMLP parameter registration (Hardcoded vs. Dynamic) & PEFT Support

### ISSUE THREAD

**Issue by akshatvishu** [snowflakedb/ArcticTraining#334]
I have been studying the Arctic Long Sequence Training (ALST) implementation, specifically the `TiledMLP` component and observing how it has been ported across the ecosystem (DeepSpeed, Liger Kernel, and Axolotl).

I noticed a specific difference in how the trainable parameters (`compute_params`) are registered for ZeRO-3 synchronization and I wanted to understand the engineering decision behind this before proposing changes.

**Reference Implementation** (Arctic, DeepSpeed, Liger): These libraries use a Hardcoded List of weights.
```py
# Logic in Arctic / DeepSpeed / Liger
compute_params = [self.down_proj.weight, self.gate_proj.weight, self.up_proj.weight]
```

**Axolotl Implementation:** The Axolotl port uses a Dynamic Scan.
```py
# Logic in Axolotl
compute_params = [p for p in self.parameters() if p.requires_grad]
```

### The Impact on LoRA / PEFT :
The "Hardcoded" approach works perfectly for the paper's original scope (Full Fine-Tuning). However, when users attempt LoRA + ZeRO-3 the hardcoded list captures only the frozen base weights and misses the trainable LoRA adapters (lora_A, lora_B).

This results in ZeRO-3 failing to synchronize the adapter gradients (because it doesn't know they are part of the compute) leading to training instability or hangs. The "Dynamic Scan" approach in Axolotl solves this by automatically picking up whatever is trainable.

Hence, my question is : Was the decision to stick with the hardcoded list in the reference implementation (and DeepSpeed) primarily for safety/strictness during full fine-tuning or simply because LoRA support wasn't in scope at the time of initial release?

 Also, if the "Dynamic Scan" approach used in Axolotl is considered safe by the authors, I would like to submit PRs to align Arctic, DeepSpeed, and Liger with this logic. This would enable out-of-the-box LoRA support for ALST workflows across the entire ecosystem without changing the behavior for Full Fine-Tuning.

cc: @sfc-gh-sbekman  (Since you are the common link across these implementations, your insight on the history here would be invaluable!)
